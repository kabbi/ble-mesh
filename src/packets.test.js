const typeSet = require('./packets');
const { hex } = require('./utils/tags');
const { parse } = require('./utils/binary')(typeSet);

describe('parsers', () => {
  describe('Advertisement', () => {
    it('parses network ad', () => {
      const data = parse('Advertisement', hex`003637d51523539028`);
      expect(data).toMatchSnapshot();
    });
    it('parses corrupted ad', () => {
      const data = parse('Advertisement', hex`023637d51523539028`);
      expect(data).toMatchSnapshot();
    });
  });
  describe('ProxyPDU', () => {
    it("parses first PDU I've ever received in my life", () => {
      const data = parse(
        'ProxyPDU',
        hex`0101003637d51523539028000000007bcfa6acdc7623bb`,
      );
      expect(data).toMatchSnapshot();
    });
  });
  describe('NetworkPDU', () => {
    it('parses strange proxy filter response', () => {
      const data = parse(
        'NetworkPDU',
        hex`4743a9ac43633a574efa2b37ab890443c04fb36764`
      );
      expect(data).toMatchSnapshot();
    })
  });
});
