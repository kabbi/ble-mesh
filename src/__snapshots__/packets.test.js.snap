// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsers Advertisement parses corrupted ad 1`] = `
Object {
  "type": 2,
}
`;

exports[`parsers Advertisement parses network ad 1`] = `
Object {
  "networkID": Object {
    "data": Array [
      54,
      55,
      213,
      21,
      35,
      83,
      144,
      40,
    ],
    "type": "Buffer",
  },
  "type": 0,
}
`;

exports[`parsers NetworkPDU parses strange proxy filter response 1`] = `
Object {
  "encryptedPart": Object {
    "data": Array [
      87,
      78,
      250,
      43,
      55,
      171,
      137,
      4,
      67,
      192,
      79,
      179,
      103,
      100,
    ],
    "type": "Buffer",
  },
  "ivi": 0,
  "nid": 71,
  "obfuscatedPart": Object {
    "data": Array [
      67,
      169,
      172,
      67,
      99,
      58,
    ],
    "type": "Buffer",
  },
}
`;

exports[`parsers ProxyPDU parses first PDU I've ever received in my life 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "auth": Object {
        "body": Object {
          "data": Array [
            0,
            54,
            55,
            213,
            21,
            35,
            83,
            144,
            40,
            0,
            0,
            0,
            0,
          ],
          "type": "Buffer",
        },
        "cmac": Object {
          "data": Array [
            123,
            207,
            166,
            172,
            220,
            118,
            35,
            187,
          ],
          "type": "Buffer",
        },
      },
      "flags": Object {
        "_reserved": 0,
        "ivUpdate": false,
        "keyRefresh": false,
      },
      "ivIndex": 0,
      "networkID": Object {
        "data": Array [
          54,
          55,
          213,
          21,
          35,
          83,
          144,
          40,
        ],
        "type": "Buffer",
      },
    },
    "type": "SecureNetwork",
  },
  "sar": "Complete",
  "type": "Beacon",
}
`;
